common/remote/rmtfile.cpp

```diff
@@ -337,9 +337,10 @@ unsigned getDaliServixVersion(const SocketEndpoint &_ep,StringBuffer &ver)
     setDafsEndpointPort(ep);
     if (ep.isNull())
         return false;
-    try {
+    try
+    {
         Owned<ISocket> socket = ISocket::connect_timeout(ep,10000);
-        return getRemoteVersion(socket,ver);
+        return getRemoteVersion(socket.getClear(), ver);
```

Comment:
Well there's a related to change to take ownership in getRemoteVersion, so won't leak...
But, I don't think that's right.

And - as it stands there are other places that call getRemoteVersion , that pass ownership of the socket and will therefore destroy it ...